sbi/inference/snle/snle_base.py

```diff
@@ -295,3 +247,82 @@ def _train(
         # Update summary.
         self._summary["epochs"].append(epoch)
         self._summary["best_validation_log_probs"].append(self._best_val_log_prob)
+
+        # Update TensorBoard and summary dict.
+        self._summarize(
+            round_=self._round, x_o=None, theta_bank=theta, x_bank=x,
+        )
+
+        # Update description for progress bar.
+        if show_train_summary:
+            print(self._describe_round(self._round, self._summary))
+
+        return deepcopy(self._neural_net)
+
+    def build_posterior(
+        self,
+        density_estimator: Optional[TorchModule] = None,
+        mcmc_method: str = "slice_np",
+        mcmc_parameters: Optional[Dict[str, Any]] = None,
+        copy_state_from: Optional[LikelihoodBasedPosteriorType] = None,
```

Comment:
same here, this could be confusing. maybe add a separate method to do this?

Reply:
If I understand correctly, you would suggest:
```python
posterior = inference.build_posterior().copy_state_from(previous_posterior)
```
The `copy_state_from()` method would then live in the `Posterior` class.

The alternative, as it is currently implemented is:
```python
posterior = inference.build_posterior(copy_state_from=previous_posterior)
```

I think I prefer the first version. What do you think?

Reply:
the first version as well.

Reply:
👍🏽 Sorry for getting at this out of order.